### DISCUSSION TITLE
Nextjs dynamic route page return 403 Forbidden error for only one specific user dynamic route

### DISCUSSION THREAD

**Discussion by rajat-devslane** [vercel/next.js#45437] [Help] [answered]
I have my route defined as `patients/[userID].tsx` in pages dir. 

When I tried to access one of the specific active userID (eg. patients/7917067d) it gave `403 Forbidden` error in prod only, In dev, it is working fine.

FYI, when I enter the wrong userId it gives 404 which is correct behaviour but don't know why only this userId which is correct gives a 403 error.

**NOTE: I am using Static Generation with Client-side data fetching**

![Screenshot from 2023-01-31 17-30-38](https://user-images.githubusercontent.com/78585850/215760036-2edf9ea3-024c-474f-931b-64078945c278.png)

EDITED:

There is a sentry error also which I think relates to the same problem:

![Screenshot from 2023-01-31 19-18-11](https://user-images.githubusercontent.com/78585850/215778593-dfe9084c-c420-401c-beb8-543ce16212be.png)

**Comment by icyJoseph** [comment]
That 403 forbidden looks nothing like Next.js' error layout, could you share what headers you get back from this request? Of course do, remove sensitive data.

**Reply by rajat-devslane** [reply]
Here is the header for the request:

![Screenshot from 2023-01-31 18-40-39](https://user-images.githubusercontent.com/78585850/215770159-6e1b839d-0843-4c6e-9bd5-abb46b212a1e.png)

**Reply by icyJoseph** [reply]
A few follow ups:

- do you see the same building locally with `next build` and `next start`?
- Are there any load balancers, or rather, who's serving this Next.js application? 
- Are you using middleware?

**Reply by rajat-devslane** [reply]
Thanks for the reply, answering your query:
* No I didn't see it when I build locally and run it using `next start`
* Not that I am aware of but this nextjs application is deployed using a docker container in google kubernative system
* Not any specific middleware only have added _middleware.ts file which has following content:
```ts
 const ContentSecurityPolicy = `
    script-src 'self' ${noEval} ${heapScriptSHA} https://cdn.heapanalytics.com https://heapanalytics.com;
    style-src 'self' 'unsafe-inline' https://fonts.gstatic.com https://fonts.googleapis.com https://heapanalytics.com;
    font-src 'self' https://fonts.gstatic.com https://fonts.googleapis.com ; 
  `;

  const response = NextResponse.next();

  response.headers.set(
    "Content-Security-Policy",
    ContentSecurityPolicy.replace(/\n/g, "")
  );
  response.headers.set("X-Frame-Options", "deny");
  response.headers.set("X-Content-Type-Options", "nosniff");
  response.headers.set("Referrer-Policy", "strict-origin");
  response.headers.set("X-XSS-Protection", "1; mode=block");

```

**Reply by rajat-devslane** [reply]
Noticed one sentry error which I think relates to the same:

![Screenshot from 2023-01-31 19-18-11](https://user-images.githubusercontent.com/78585850/215778281-bb9ce3d2-29c8-47fc-9f94-479e99c723df.png)

**Reply by icyJoseph** [reply]
Aha now that maybe something to work with, the question is why wouldn't it trip locally. We have to first assume there's some network exchange that's failing. Looking at the name of the module, I am afraid this might be client side, and it is just reading that the transition into the page failed because the HEAD request (pre-flight) errored out with 403, but I'll take a look.

**Reply by icyJoseph** [reply]
Ah, look at what I found, https://github.com/vercel/next.js/pull/37490

What Next.js version do you have?

**Reply by rajat-devslane** [reply]
I am using `12.1.0`

**Reply by icyJoseph** [reply]
I see, I think it might be worth upgrading and seeing if you still hit the error. I do recall there were some bugs around those versions, but I can't really point you in the right direction (to see if the bugs might be related to your case). Since tons of people report issues, it is hard to track an old issue :/

**Reply by rajat-devslane** [reply]
@icyJoseph I had the plan to upgrade it to the latest version but since it has major changes so I didn't upgrade it now as it was breaking my existing code. 

Will update you once I upgrade the nextjs. Thanks for the help.

**Reply by icyJoseph** [reply]
You can upgrade to the latest 12.x.x version though. Not sure if it is 12.2.6, and see if the bug is still present, or at least takes another shape.

**Reply by rajat-devslane** [reply]
Sure I'll check that.

**Comment by rajat-devslane** [comment, marked as answer]
@icyJoseph Upgrading the nextjs package to 12.3.3 did the work. Thanks for your effort. Appreciate it.

**Comment by rajat-devslane** [comment]
@icyJoseph I just noticed this issue in a different way. When I go through the link to that route it works fine but when I directly enter that URL in a browser or refresh the page it gives `403 Forbidden` error as just before.

**Comment by goldendilemma** [comment]
If somebody else stumbles upon this. I had this problem when using `.htaccess` with the directory indexing disabled `Options -Indexes`.

Fixed it by adding new default index files (index.html is standard) and in my case next generated a file in subfolder with a file called `[[...adventureId]].html`

Adding the following line to the `.htaccess` file fixed it for me
```
# [[...*]] is necessary for dynamically matched routes in static generation of next
DirectoryIndex index.html [[...adventureId]].html
```

**Comment by monirsaikat** [comment]
This is still not working with next.js 15.1.4 LOL
